web/src/app/core/services/repository.service.ts

```diff
@@ -59,7 +59,10 @@ export class RepositoryService {
     let rating: number = 0;
     const issuePoints: number = repo.issues.length > 0 ? this.getPoints(repo.issues[0].createdOn) : 0;
     const releasesPoints: number = repo.releases.length > 0 ? this.getPoints(repo.releases[0].createdOn) : 0;
-    rating = (issuePoints + releasesPoints) / 2;
+    const milestonesPoints: number = repo.milestones.length > 0 ? this.getPoints(new Date(repo.milestones[0].updatedAt)) : 0;
+    const urlPoints: number = repo.url ? 100 : 0;
+    const descriptionPoints: number = repo.description ? 100 : 0;
+    rating = (issuePoints + releasesPoints + milestonesPoints + urlPoints + descriptionPoints) / 5;
```

Comment:
Looks good 👍 . 

May be as we have many it might be cleaner to write a loop? Something like...

```typescript
const checks: number[] = [];
checks.push(repo.issues.length > 0 ? this.getPoints(repo.issues[0].createdOn) : 0);
checks.push(repo.releases.length > 0 ? this.getPoints(repo.releases[0].createdOn) : 0);		    
checks.push(repo.releases.length > 0 ? this.getPoints(repo.releases[0].createdOn) : 0);
checks.push(repo.milestones.length > 0 ? this.getPoints(new Date(repo.milestones[0].updatedAt)) : 0);
checks.push(repo.url ? 100 : 0);
checks.push(repo.description ? 100 : 0);

rating = checks.reduce((total, current) => total + current, 0) / checks.length
```

Reply:
do we need to add that in case if any of these fields doesn't exist

Reply:
I think the flow is the same, it just seems easier for the last line, so we don't need to keep adding a `+ newitem` and then increment the divide. 

All that would need to be done for a new calculation is push to the `checks` array and the accumulation and divide would just figure itself out

Reply:
sure !! I have updated the method with above logic.